### ISSUE TITLE
ember-cli allows you to create helpers without hyphens without warning you they will not be automatically loaded

### ISSUE THREAD

**Issue by ac-pevans** [ember-cli/ember-cli#3286]
In the documentation on [using modules](http://www.ember-cli.com/using-modules/) at ember-cli.com, there is:

> Custom Handlebars helpers are one of the ways that you can use the same HTML multiple times in your web application. Registering your custom helper allows it to be invoked from any of your Handlebars templates. Custom helpers are located under app/helpers. If your custom helper contains a dash(upper-case, reverse-word, etc.), it will be found and loaded automatically by the resolver.
...
> Limiting automatically-loaded helpers to those that contain dashes is an explicit decision made by Ember. It helps disambiguate properties from helpers, and helps mitigate the performance hit of helper resolution for all bindings. The other loading option is to define only the function used by the helper and to load it explicitly [...]

It's fine to let people create helpers without hyphens, but if they do, it would be helpful to print a warning at that time to let people know that they will need to make that explicit load for their helper to be recognized.

**Comment by stefanpenner** [comment]
we could make them load automatically...

@ember-cli/owners thoughts on them loading automatically?

**Comment by rwjblue** [comment]
@stefanpenner - I'm not opposed to loading automatically, but I am opposed to loading them via an initializer.  Loading them via an initializer makes it more difficult to integration test (multiple units but no started app) a component when it may use a third party component.

**Comment by stefanpenner** [comment]
i believe we should add `Resolver.resolveAll(type)` 

example:

```js
resolver.resolveAll('helpers').forEach(function(name, helper) {
  // ... registration
});
```

**Comment by codefox421** [comment]
:+1: at least for the warning. Silently failing is never a good thing.

**Comment by IngageStroliaC** [comment]
I just got dinged by this one. 

:+1: from me to fix this issue as well, although, a warning would likely need to come on the Ember side, not ember-cli, IMHO. Perhaps one solution might be to auto prefix single word helpers with something like "do-" or something.

**Comment by stefanpenner** [comment]
https://github.com/stefanpenner/ember-cli-auto-register-helpers if you want the auto-loading functionality

**Comment by IngageStroliaC** [comment]
Thanks Stefan, that did the trick for me.

**Comment by stefanpenner** [comment]
Not really a bug, although a gotcha. If someone wants to pursue this above ideas should become an RFC and a potential reference implementation.

**Comment by balinterdi** [comment]
It seems that Ember CLI 1.13.8 (Ember 1.13.9) no longer enforces this. I did `ember generate helper capitalize` and although I did receive the warning about my helper not being automatically resolved, I used `{{capitalize band.name}}` in the template and it worked without any problems (tried with `cap`, too). 

Has something changed recently and now helpers without dashes will be looked up, too? (I don't have the ember-cli-auto-register-helpers addon installed)
